### PR TITLE
[CUR-917] Close button on lot picker for subject phase

### PR DESCRIPTION
**Pull request by sonalisin**
## Description

Music year: 1951

- Add a close button to the lot picker for both subject and phase modals

## How to test

1. Go to https://deploy-preview-2951--oak-web-application.netlify.thenational.academy/teachers/curriculum
2. Click on the subject selector
3. You should see the close button in the top right corner
4. Click on the button and the modal should close
5. Repeat for phase selector
6. Shouldn't impact subject or phase selection

## Should look like
Screenshot 2024-10-31 at 15 40 45
Screenshot 2024-10-31 at 15 40 55

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumModalCloseButton/CurriculumModalCloseButton.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+type CurriculumModalCloseButtonProps = {
+  onClose: () => void;
+};
+
+export const CurriculumModalCloseButton: FC<
+  CurriculumModalCloseButtonProps
+> = ({ onClose }) => {
+  return (
+    <Box $position={"absolute"} $top={[16, 32]} $right={[16, 32]}>
```
Comment (orangemug) [inline src/components/CurriculumComponents/CurriculumModalCloseButton/CurriculumModalCloseButton.tsx:14]: I don't think this should live in the component. Generally you want positioning to stay outside the component as components should be able to live in other places.

So put the `<Box/>` into the two places in `./src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx`

File: src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ import OwaLink from "@/components/SharedComponents/OwaLink";
 import Box from "@/components/SharedComponents/Box";
 import BoxBorders from "@/components/SharedComponents/SpriteSheet/BrushSvgs/BoxBorders/BoxBorders";
 import Flex from "@/components/SharedComponents/Flex.deprecated";
+// import IconButtonFocusVisible from "@/components/CurriculumComponents/UnitsTabSidebar";
```
Comment (orangemug) [inline src/components/SharedComponents/SubjectPhasePicker/SubjectPhasePicker.tsx:22]: Old code

**Review by HelenHarlow — COMMENTED** [review]
Looks good! Would it be possible to match the button size to the size used in our unit and navigation modals? When I checked I believe these are 30px although that seems like an odd size. 
It also looks like the close icon is the last focussable element in the modal, is it possible to make it the first focussable element?

**Comment by rightsaidjames** [comment]
@sonalisin I've tested this on desktop and mobile, with both a keyboard and screen reader. Looks good, but I want to wait for the requested changes above to be implemented before I do my final testing.

I notice that @HelenHarlow's request to move the close button to the start of the focus order conflicts with my suggestion to keep it at the end, but having played around a little bit (and thinking about it some more), putting it at the start does make more sense for consistency purposes and also to help screen reader users find the close button more easily. Sorry for flip-flopping on this 😅

**Review by orangemug — APPROVED** [review]

**Review by HelenHarlow — APPROVED** [review]
Looks good!

**Review by rightsaidjames — APPROVED** [review]
Looks good - all acceptance criteria met, buttons are in the correct place in the focus order as per @HelenHarlow's amendments, requested changes made.

Screenshot 2024-11-11 at 09 41 09
Screenshot 2024-11-11 at 09 41 15
Screenshot 2024-11-11 at 09 41 24
Screenshot 2024-11-11 at 09 41 44
Screenshot 2024-11-11 at 09 41 54

**Pull request merged**